ledger/src/shred.rs

```diff
@@ -536,25 +537,25 @@ impl Shred {
 
     // Returns the block index within the erasure coding set.
     fn erasure_block_index(&self) -> Option<usize> {
-        let index = self.index().checked_sub(self.fec_set_index())?;
-        let index = usize::try_from(index).ok()?;
         match self.shred_type() {
-            ShredType::Data => Some(index),
+            ShredType::Data => {
+                let index = self.index().checked_sub(self.fec_set_index())?;
+                usize::try_from(index).ok()
+            }
             ShredType::Code => {
-                // TODO should use first_coding_index once position field is
-                // populated.
                 // Assert that the last shred index in the erasure set does not
                 // overshoot u32.
                 self.fec_set_index().checked_add(u32::from(
-                    self.coding_header
-                        .num_data_shreds
-                        .max(self.coding_header.num_coding_shreds)
-                        .checked_sub(1)?,
+                    self.coding_header.num_data_shreds.checked_sub(1)?,
```

Comment:
really want to rename `fec_set_index()` to something more explicit like `first_data_shred_index_in_fec_set()`, I have to look it up every time because I have the memory of a 🐟

Reply:
Well it is an index meant to identify erasure batches within the slot.
Coincidentally, It has become an implementation detail that it is also equal to the first coding shred index in the batch, but the code outside this file should not care about that, and I am not sure that has always been the case or stays so going forward.